### PR TITLE
Applied rustfmt uniformly across codebase

### PR DESCRIPTION
**Pull request by dominic-mulligan-arm**
Rather than applying `rustfmt` on a file-by-file basis, as files are modified, which pollutes PRs with needless noise, every file in the project has now been formatted.

- [x] Fixed top-level "utility" build targets in the project Makefile, like `fmt`, `clean`, etc.  Made sure every project component is covered, and pushed the targets down, recursively, into lower-level Makefiles for project components that use them.
- [x] Added `fmt-check` build target which fails if a Rust file is not properly formatted with `rustfmt`.
- [x] Applied `rustfmt` to every project file.
- [x] Added `fmt-check` as a new CI workflow check (in the `compile-and-check-sdk` step).

### CODE REVIEW

File: sdk/data-generators/shamir-secret-sharing/src/main.rs [hunk 1]
```diff
@@ -12,127 +12,70 @@
...
+    0xc4, 0x57, 0xf9, 0x10, 0x30, 0x50, 0xf0, 0x0b, 0x1d, 0x27, 0x69, 0xbb, 0xd6, 0x61, 0xa3, 0xfe,
+    0x19, 0x2b, 0x7d, 0x87, 0x92, 0xad, 0xec, 0x2f, 0x71, 0x93, 0xae, 0xe9, 0x20, 0x60, 0xa0, 0xfb,
+    0x16, 0x3a, 0x4e, 0xd2, 0x6d, 0xb7, 0xc2, 0x5d, 0xe7, 0x32, 0x56, 0xfa, 0x15, 0x3f, 0x41, 0xc3,
+    0x5e, 0xe2, 0x3d, 0x47, 0xc9, 0x40, 0xc0, 0x5b, 0xed, 0x2c, 0x74, 0x9c, 0xbf, 0xda, 0x75, 0x9f,
+    0xba, 0xd5, 0x64, 0xac, 0xef, 0x2a, 0x7e, 0x82, 0x9d, 0xbc, 0xdf, 0x7a, 0x8e, 0x89, 0x80, 0x9b,
+    0xb6, 0xc1, 0x58, 0xe8, 0x23, 0x65, 0xaf, 0xea, 0x25, 0x6f, 0xb1, 0xc8, 0x43, 0xc5, 0x54, 0xfc,
+    0x1f, 0x21, 0x63, 0xa5, 0xf4, 0x07, 0x09, 0x1b, 0x2d, 0x77, 0x99, 0xb0, 0xcb, 0x46, 0xca, 0x45,
+    0xcf, 0x4a, 0xde, 0x79, 0x8b, 0x86, 0x91, 0xa8, 0xe3, 0x3e, 0x42, 0xc6, 0x51, 0xf3, 0x0e, 0x12,
+    0x36, 0x5a, 0xee, 0x29, 0x7b, 0x8d, 0x8c, 0x8f, 0x8a, 0x85, 0x94, 0xa7, 0xf2, 0x0d, 0x17, 0x39,
+    0x4b, 0xdd, 0x7c, 0x84, 0x97, 0xa2, 0xfd, 0x1c, 0x24, 0x6c, 0xb4, 0xc7, 0x52, 0xf6,
```
Comment (geky) [inline sdk/data-generators/shamir-secret-sharing/src/main.rs:70]: Are you able to mark these as `#[rustfmt::skip]`? This formatting looses the info that it's two separate 255-byte blocks.

File: sdk/rust-examples/shamir-secret-sharing/src/main.rs [hunk 2]
```diff
@@ -28,119 +28,67 @@
...
+    0xc4, 0x57, 0xf9, 0x10, 0x30, 0x50, 0xf0, 0x0b, 0x1d, 0x27, 0x69, 0xbb, 0xd6, 0x61, 0xa3, 0xfe,
+    0x19, 0x2b, 0x7d, 0x87, 0x92, 0xad, 0xec, 0x2f, 0x71, 0x93, 0xae, 0xe9, 0x20, 0x60, 0xa0, 0xfb,
+    0x16, 0x3a, 0x4e, 0xd2, 0x6d, 0xb7, 0xc2, 0x5d, 0xe7, 0x32, 0x56, 0xfa, 0x15, 0x3f, 0x41, 0xc3,
+    0x5e, 0xe2, 0x3d, 0x47, 0xc9, 0x40, 0xc0, 0x5b, 0xed, 0x2c, 0x74, 0x9c, 0xbf, 0xda, 0x75, 0x9f,
+    0xba, 0xd5, 0x64, 0xac, 0xef, 0x2a, 0x7e, 0x82, 0x9d, 0xbc, 0xdf, 0x7a, 0x8e, 0x89, 0x80, 0x9b,
+    0xb6, 0xc1, 0x58, 0xe8, 0x23, 0x65, 0xaf, 0xea, 0x25, 0x6f, 0xb1, 0xc8, 0x43, 0xc5, 0x54, 0xfc,
+    0x1f, 0x21, 0x63, 0xa5, 0xf4, 0x07, 0x09, 0x1b, 0x2d, 0x77, 0x99, 0xb0, 0xcb, 0x46, 0xca, 0x45,
+    0xcf, 0x4a, 0xde, 0x79, 0x8b, 0x86, 0x91, 0xa8, 0xe3, 0x3e, 0x42, 0xc6, 0x51, 0xf3, 0x0e, 0x12,
+    0x36, 0x5a, 0xee, 0x29, 0x7b, 0x8d, 0x8c, 0x8f, 0x8a, 0x85, 0x94, 0xa7, 0xf2, 0x0d, 0x17, 0x39,
+    0x4b, 0xdd, 0x7c, 0x84, 0x97, 0xa2, 0xfd, 0x1c, 0x24, 0x6c, 0xb4, 0xc7, 0x52, 0xf6,
```
Comment (geky) [inline sdk/rust-examples/shamir-secret-sharing/src/main.rs:83]: Are you able to mark these as `#[rustfmt::skip]`? This formatting looses the info that it's two separate 255-byte blocks.

**Review by mathias-arm — DISMISSED** [review]

File: sdk/data-generators/shamir-secret-sharing/src/main.rs (hunk 1 above)
Comment (dominic-mulligan-arm) [reply sdk/data-generators/shamir-secret-sharing/src/main.rs:72]: Done.

File: sdk/rust-examples/shamir-secret-sharing/src/main.rs (hunk 2 above)
Comment (dominic-mulligan-arm) [reply sdk/rust-examples/shamir-secret-sharing/src/main.rs:85]: Done.

**Review by mathias-arm — DISMISSED** [review]

**Comment by egrimley-arm** [comment]
You could get rid of trailing spaces while you're at it:
```
perl -i -pe 's/[ \t]*$//;' `find -name target -prune -o -name \*.rs -print`
```

**Comment by nspin** [comment]
Eventually we might want to de-duplicate those `sdk/**/Makefile`s with a workspace and some higher-level Make code.

**Review by egrimley-arm — APPROVED** [review]

**Comment by dominic-mulligan-arm** [comment]
+1+1 = +1 reached, merging.

**Pull request merged**